torch/distributed/tensor/placement_types.py

```diff
@@ -89,39 +89,22 @@ def _split_tensor(
 
         # chunk tensor over dimension `dim` into n slices
         tensor_list = list(torch.chunk(tensor, num_chunks, dim=self.dim))
-        num_empty_tensors = num_chunks - len(tensor_list)
```

Comment:
I'm simplifying the implementation of this `Shard._split_tensor` method, without changing the behavior (except for when we return an empty list for pad_sizes), and then I'll reuse the same structute for `_StridedShard._split_tensor` below.